File: tests/cases/basic/__snapshots__/while-loop.lua
local ____exports = {}
local function countDown(n)
    while true do
        do
            if not (n > 0) then
                break
            end
            n = n - 1
        end
        ::____continue_0::
    end
    return n
end
local function sumTo(n)
    local i = 0
    local total = 0
    while true do
        do
            if not (i < n) then
                break
            end
            i = i + 1
            total = total + i
        end
        ::____continue_1::
    end
    return total
end
local function doAtLeastOnce(n)
    local i = 0
    while true do
        do
            i = i + 1
        end
        ::____continue_2::
        if not (i < n) then
            break
        end
    end
    return i
end
local function whileBreak()
    local i = 0
    while true do
        do
            if not true then
                break
            end
            if i >= 5 then
                break
            end
            i = i + 1
        end
        ::____continue_3::
    end
    return i
end
local function whileContinue()
    local i = 0
    local evens = 0
    while true do
        do
            if not (i < 10) then
                break
            end
            i = i + 1
            if i % 2 == 1 then
                goto ____continue_4
            end
            evens = evens + 1
        end
        ::____continue_4::
    end
    return evens
end
return ____exports

File: tests/cases/basic/while-loop.ts
function countDown(n: number) {
  while (n > 0) {
    n--;
  }
  return n;
}

function sumTo(n: number) {
  let i = 0;
  let total = 0;
  while (i < n) {
    i++;
    total += i;
  }
  return total;
}

function doAtLeastOnce(n: number) {
  let i = 0;
  do {
    i++;
  } while (i < n);
  return i;
}

function whileBreak() {
  let i = 0;
  while (true) {
    if (i >= 5) break;
    i++;
  }
  return i;
}

function whileContinue() {
  let i = 0;
  let evens = 0;
  while (i < 10) {
    i++;
    if (i % 2 === 1) continue;
    evens++;
  }
  return evens;
}
